php/Block.php

```diff
@@ -63,66 +63,179 @@ public function register_block() {
+		 * maps the post type's label name (not slug), and the number of
+		 * published posts within that post type so we can easily display it
+		 * below.
+		 */
+
+		// Array to store our new values in.
+		$public_post_type_counts = [];
+
+		// Get all public post type objects.
+		$post_type_objects = get_post_types(  [ 'public' => true ], 'objects' );
+
+		// Validate the values.
+		if (
+			! empty( $post_type_objects )
+			&& is_array( $post_type_objects )
+		) {
+
+			// Loop through each result and store the label and count mapped to
+			// the slug.
+			foreach ( $post_type_objects as $post_type_object ) {
+				/**
+				 * @var array $public_post_type_counts = [
+				 *     'posts' => [
+				 *         'label' => 'Posts',
+				 *         'count' => 5,
+				 *      ],
+				 *      'pages' => [
+				 *        'label' => 'Pages',
+				 *        'count' => 3,
+				 *      ],
+				 * ]
+				 */
+				$public_post_type_counts[ $post_type_object->name ] = [
+					'label' => $post_type_object->labels->name ?? '',
+					'count' => wp_count_posts( $post_type_object->name )->publish ?? 0,
+				];
+			}
+		}
+
+		/**
+		 * Query for up to 5 posts & pages published between 9:00 and 17:00,
+		 * with the tag (slug) `foo`, and category (name) `baz`.
+		 */
+		$posts_list_query = new WP_Query(
+			[
+				'post_type'              => [ 'post', 'page' ],
+				'post_status'            => 'publish',
+				'fields'                 => 'ids',
+				'update_post_meta_cache' => false, // We're not displaying any post meta.
+				'update_post_term_cache' => false, // Or terms.
+				'posts_per_page'         => 6, // Actual number of posts we want (5), plus one for our current post.
+				'date_query'             => [
+					[
+						'hour'   => 9,
+						'compare'=> '>=',
+					],
+					[
+						'hour'   => 17,
+						'compare'=> '<=',
+					]
+				],
+				'tax_query' => [
+					[
+						'taxonomy' => 'post_tag',
+						'field'    => 'slug',
+						'terms'    => 'foo',
+					],
+					[
+						'taxonomy'=> 'category',
+						'field'   => 'name',
+						'terms'   => 'baz',
+					]
+				],
+			]
+		);
+
+		/**
+		 * If our current post (`get_the_ID()`) is returned in the query results,
+		 * remove it. This mimics the functionality of using the `post__not_in`
+		 * argument on the query.
+		 *
+		 * @see https://docs.wpvip.com/technical-references/code-quality-and-best-practices/using-post__not_in/
+		 */
+		if ( in_array( $current_post_id, $posts_list_query->posts, true ) ) {
+
+			// Remove the current post id by using `array_diff`, then reset the
+			// keys using `array_values` to ensure no gaps.
+			$posts_list_query->posts = array_values( array_diff( $posts_list_query->posts, [ $current_post_id ] ) );
+
+			// Update the post count and found post values now that the current
+			// post ID has been removed.
+			$posts_list_query->post_count    = $posts_list_query->post_count - 1;
+			$posts_list_query->found_posts   = $posts_list_query->found_posts - 1;
+		}
```

Comment:
I like this approach better than writing an exclusion for the current post in the output loop. It's a bit cleaner to follow in both locations, and it keeps the data integrity of the WP_Query a little closer to what we actually intend (i.e. if we'd used `post__not_in`).

Also, this code isn't "fool-proof". For example, it doesn't recompute the `max_num_pages` value, but we're only working with 5 (6) posts anyway, so I didn't bother.

It's a couple of years old and outdated at this point (there are some bugs that weren't addressed in the fix), but I did write a class called `Unique_WP_Query` which handles specific instances of the `post__not_in` problem.

https://gist.github.com/jameswburke/d0776d742ab74c469cd8af8dacd916fc